msgraph_async/client/tests/test_client.py

```diff
@@ -830,3 +831,23 @@ async def test_get_domain(self):
         i = self.get_instance()
         res, status = await i.get_domain(TestClient._domain_id, token=TestClient._token)
         self.assertEqual(status, HTTPStatus.OK)
+
+    async def test_list_service_principles(self):
+        i = self.get_instance()
+        res, status = await i.list_service_principles(token=TestClient._token)
+        self.assertEqual(status, HTTPStatus.OK)
+
+    async def test_get_service_principle(self):
+        i = self.get_instance()
+        res, status = await i.get_service_principle(TestClient._service_principle_id, token=TestClient._token)
+        self.assertEqual(status, HTTPStatus.OK)
+
+    async def test_get_service_principle_with_filter(self):
+        i = self.get_instance()
+        microsoft_discovery_service_app_id = "'6f82282e-0070-4e78-bc23-e6320c5fa7de'"
```

Comment:
app id should be defined in the configuration file, and not be visible to everybody as part of this repo